master/buildbot/db/logs.py

```diff
@@ -329,6 +329,8 @@ def thdcompressLog(conn):
             q = sa.select([sa.func.sum(sa.func.length(tbl.c.content))])
             q = q.where(tbl.c.logid == logid)
             newsize = conn.execute(q).fetchone()[0]
+            if newsize is None:
+                newsize = totlength
```

Comment:
I too don't know enough of the possible values that code can generate...

>>> None or 5
5
>>> 0 or 5
5
>>> 0 or -5
-5
>>> None or -5
-5